storage-proto/proto/entries.proto

```diff
@@ -0,0 +1,15 @@
+syntax = "proto3";
+
+package solana.storage.Entries;
+
+message Entries {
+    repeated Entry entries = 1;
+}
+
+message Entry {
+    uint32 index = 1;
+    uint64 num_hashes = 2;
+    bytes hash = 3;
+    uint64 num_transactions = 4;
+    uint32 starting_transaction_index = 5;
```

Comment:
I know this was technically added in https://github.com/solana-labs/solana/pull/33995, but remind me, the `starting_transaction_index` is for convenience and not strictly critical right?

For example, suppose we have following entries:
```
[
    Entry0 = Entry { ... , num_transactions = 2, starting_transaction_index = 0 }.
    Entry1 = Entry { ... , num_transactions = 0, starting_transaction_index = 2 }.
    Entry2 = Entry { ... , num_transactions = 1, starting_transaction_index = 2 }
]
```
For `EntryN`, the starting_transaction_index is the sum of the `num_transactions` for `[Entry0, EntryN-1]`.

And to be clear, I'm not proposing we change it, just asking for my understanding

Reply:
Correct, it's a convenience. Without that field, if you want to determine the set of transactions in a particular entry, you have to iterate through all the previous entries in the block.